src/game/overlay.c

```diff
@@ -113,40 +193,27 @@ static void Overlay_GetBarLocation(
     m_BarOffsetY[bar_location] += height + bar_spacing;
 }
 
-static void Overlay_DrawBar(int32_t value, int32_t value_max, int32_t bar_type)
+static void Overlay_DrawBar(BAR_INFO *draw_bar)
 {
-    static int32_t blink_counter = 0;
     const int32_t percent_max = 100;
 
-    if (value < 0) {
-        value = 0;
-    } else if (value > value_max) {
-        value = value_max;
+    if (draw_bar->value < 0) {
+        draw_bar->value = 0;
+    } else if (draw_bar->value > draw_bar->max_value) {
+        draw_bar->value = draw_bar->max_value;
     }
-    int32_t percent = value * 100 / value_max;
+    int32_t percent = draw_bar->value * 100 / draw_bar->max_value;
```

Comment:
We can use `CLAMP` value. Also since we duplicate the code from above, maybe it's good to introduce a function for this `Overlay_BarGetPercent(BAR_INFO *bar)`